### DISCUSSION TITLE
Is it not possible to use IP Adapter with DirectML?

### DISCUSSION THREAD

**Discussion by kofteburger** [lshqqytiger/stable-diffusion-webui-directml#304] [General]
I've tried to use the IP Adapter Controlnet Model with this port of the WebUI but it failed. 
As I understand it, IP adapter uses Clipvisiondetector which only supports CUDA or CPU. 
Would it be possible to force CPU only just for IP Adapter model? 

>   RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.

**Comment by kofteburger** [comment]
For those searching for this. I didn't find a way to get IP Adapter working with Automatic1111 DirectML it works on ComfyUI DirectML using this [custom node.](https://github.com/cubiq/ComfyUI_IPAdapter_plus) It runs without controlnet and the clip vision part is handled by the cpu.

**Reply by DrMacabre** [reply]
do you have a workflow to share? i tried with the one available on comfyUi ipadapter plus page but it's throwing errors and get stuck on the ksampler node.

**Reply by kofteburger** [reply]
I had since caved in and installed Linux but as I recall it worked with the default workflow available on the repo page.
